### PR TITLE
Fix error handling in migrate-app command

### PR DESCRIPTION
**Pull request by kemmerle**
## Description and Context
In this PR, I've fixed a bug with our error handling in the `migrate-app` command. In certain scenarios, the `error` returned from the API is a property on the response object (as in `error.error`). In others, the API simply returns the error. We now account for both instances. 

## Screenshots

Before 😢 :

Screenshot 2024-06-06 at 11 04 27 AM

After 😄 :

Screenshot 2024-06-06 at 11 04 55 AM

## TODO

- [x] Address any feedback

## Who to Notify

@mendel-at-hubspot @brandenrodgers @camden11 @joe-yeager

### CODE REVIEW

**Review by brandenrodgers — APPROVED** [review]

File: packages/cli/commands/project/migrateApp.js [hunk 1]
```diff
@@ -168,12 +168,16 @@ exports.handler = async options => {
...
     }
-  } catch (e) {
+  } catch (error) {
     SpinniesManager.fail('migrateApp', {
       text: i18n(`${i18nKey}.migrationStatus.failure`),
       failColor: 'white',
     });
-    logApiErrorInstance(e.error, new ApiErrorContext({ accountId }));
+    if (error.error) {
+      logApiErrorInstance(error.error, new ApiErrorContext({ accountId }));
```
Comment (brandenrodgers) [inline packages/cli/commands/project/migrateApp.js:177]: Not a blocker, but you could shorten this to `error.error || error` in logApiErrorInstance() if you want to cut down on lines of code

Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js]: Nice! Addressed in https://github.com/HubSpot/hubspot-cli/pull/1089/commits/210549adfc36170bd25df9d6dcb2abde6036b3b8
